bioio/array_like_reader.py

```diff
@@ -0,0 +1,470 @@
+
+    channel_names: Optional[Union[List[str], List[List[str]]]]
+        A list of string channel names to be applied to all array(s) or a
+        list of lists of string channel names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (create OME channel IDs for names for single or multiple arrays)
+
+    physical_pixel_sizes: Optional[
+        Union[List[float], Dict[str, float], PhysicalPixelSizes]
+    ]
+        A specification of this image's physical pixel sizes. Can be provided as
+        a list, dict or PhysicalPixelSizes object. If a list is passed, the assumed
+        order is [Z, Y, X]. If a dict is passed, it must contain "Z", "Y" and "X"
+        as keys.
+        Default: None
+
+    Raises
+    ------
+    exceptions.ConflictingArgumentsError
+        Raised when the number of scenes provided is different from the number of items
+        provided to the metadata parameters if as a list.
+
+    exceptions.ConflictingArgumentsError
+        Raised when channel_names is provided but the channel dimension was
+        either not guessed or not provided in dim_order.
+
+    ValueError
+        Provided dim_order string or channel_names are not the same length as
+        the number of dimensions or the size of the channel dimensions for the array at
+        the matching index.
+
+    Notes
+    -----
+    If you want to combine multiple numpy ndarrays or dask arrays with xarray DataArrays
+    and attach metadata to all non-xarray.DataArrays, you can do so, the metadata for
+    the xarray DataArray scenes will simply be ignored if it the non-xarray defaults.
+
+    In such cases, it is recommended that you provide metadata for those scenes as
+    None. I.E.
+
+    >>> some_metadata_attached_xr = ...
+    ... some_np = ...
+    ... some_dask = ...
+    ... reader = ArrayLikeReader(
+    ...     image=[some_metadata_attached_xr, some_np, some_dask],
+    ...     dim_order=[None, "CTYX", "ZTYX"],
+    ...     channel_names=[None, ["A", "B", C"], None],
+    ... )
+
+    Will create a three scene ArrayLikeReader with the metadata and coordinate
+    information filled xarray DataArray as the first scene, a numpy array with CTYX as
+    the dimensions and ["A", "B", "C"] as the channel names, and a dask array with ZTYX
+    as the dimensions and no channel names (as there is no channel dimension).
+    """
+
+    _scenes: Optional[Tuple[str, ...]] = None
+
+    @staticmethod
+    def _is_supported_image(  # type: ignore
+        image: Union[List[MetaArrayLike], MetaArrayLike], *args: Any, **kwargs: Any
+    ) -> bool:
+        if isinstance(image, list):
+            return all(
+                isinstance(scene, (np.ndarray, da.Array, xr.DataArray))
+                for scene in image
+            )
+
+        return isinstance(image, (np.ndarray, da.Array, xr.DataArray))
+
+    @staticmethod
+    def _guess_dim_order(shape: Tuple[int, ...]) -> str:
```

Comment:
Just curious. I think this may be the third time we have this code in multiple places? (Base) TiffReader, this one, and somewhere else?? Should we centralize this to a standard function that is called `guess_dims_from_shape` that all of these different Readers call?